akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala

```diff
@@ -703,7 +908,45 @@ final class ClusterReceptionist(pubSubMediator: ActorRef, settings: ClusterRecep
       }
 
     case _: MemberEvent ⇒ // not of interest
+
+    case GetClusterClients ⇒
+      sender() ! ClusterClients(clientInteractions.keySet)
+
+    case CheckDeadlines ⇒
+      clientInteractions = clientInteractions.filter {
+        case (_, failureDetector) ⇒
+          failureDetector.isAvailable
+      }
+      publishClientsUnreachable()
+  }
+
+  def updateClientInteractions(client: ActorRef): Unit =
+    clientInteractions.get(client) match {
+      case Some(failureDetector) ⇒
+        failureDetector.heartbeat()
+      case None ⇒
+        val failureDetector = new DeadlineFailureDetector(acceptableHeartbeatPause, heartbeatInterval)
+        clientInteractions = clientInteractions + (client -> failureDetector)
+    }
+
+  def publishClientsUp(): Unit = {
+    val publishableClients = clientInteractions.keySet.to[HashSet]
+    (publishableClients -- clientsPublished)
+      .foreach { c ⇒
+        log.info("Received new contact from [{}]", c.path)
```

Comment:
OK, debug it is then.

> On 9 May 2016, at 6:27 PM, Patrik Nordwall <notifications@github.com> wrote:
> 
> In akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala <https://github.com/akka/akka/pull/20462#discussion_r62463994>:
> 
> > +  }
> > +
> > +  def updateClientInteractions(client: ActorRef): Unit =
> > +    clientInteractions.get(client) match {
> > +      case Some(failureDetector) ⇒
> > +        failureDetector.heartbeat()
> > +      case None ⇒
> > +        val failureDetector = new DeadlineFailureDetector(acceptableHeartbeatPause, heartbeatInterval)
> > +        clientInteractions = clientInteractions + (client -> failureDetector)
> > +    }
> > +
> > +  def publishClientsUp(): Unit = {
> > +    val publishableClients = clientInteractions.keySet.to[HashSet]
> > +    (publishableClients -- clientsPublished)
> > +      .foreach { c ⇒
> > +        log.info("Received new contact from [{}]", c.path)
> always difficult to pick log level, but I think it can be too verbose on server side, given that there might be 1000s of clients coming and going
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub <https://github.com/akka/akka/pull/20462/files/22b7e59e609b89af8a35659692e7f8612bb66c33#r62463994>